Main.java

```diff
@@ -9,7 +10,19 @@
    * @param args the command line arguments
    */
   public static void main(String[] args) {
-    // TODO code application logic here
-    
+    Scanner input = new Scanner(System.in);
+    // asking for all the points and initiating their variables
+    System.out.println("Please enter in the x coordinate of the first point");
+    int x1 = input.nextInt();
+    System.out.println("Please enter in the y coordinate of the first point");
+    int y1 = input.nextInt();
+    System.out.println("Please enter in the x coordinate of the second point");
+    int x2 = input.nextInt();
+    System.out.println("Please enter in the y coordinate of the second point");
+    int y2 = input.nextInt();
+    // calculating the slope and initiating variable slope
+    int slope = (y2 - y1) / (x2 - x1);
```

Comment:
Can you have a point where the x or y value has a decimal place? These should probably be doubles instead of ints. We have a hint to this because in the sample output it shows the slope as 2.0 instead of just 2. Otherwise, this is pretty good looking code.